### PR TITLE
[terra-collapsible-menu-view] Flicker issue when wrapped with clinical header

### PR DESCRIPTION
**Pull request by MadanKumarGovindaswamy**
### Summary

**What was changed:**

Updated to calculate the available space from the collapsible menu container div when collapsible menu view component is wrapped with clinical header component.

**Why it was changed:**

When a file path with a small label is added dynamically inside a clinical header it starts to flicker.

### Testing

This change was tested using:

- [x] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [x] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [x] Functional review

### Additional Details

https://github.com/cerner/terra-framework/assets/132886427/55f8cb5c-6598-4a49-b551-d717879bb7a1

**This PR resolves:**

UXPLATFORM-10099 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-framework-docs/src/terra-dev-site/test/file-path/FilePathwithClinicalHeader.test.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  return (
+    <>
+      <button id="add-path" type="button" onClick={handleClick}>Click Me</button>
+      <ActionHeader text={<FilePath items={items} />} />
+    </>
+  );
+};
+
+export default FilePathwithClinicalHeader;
```
Comment (supreethmr) [inline packages/terra-framework-docs/src/terra-dev-site/test/file-path/FilePathwithClinicalHeader.test.jsx:32]:
```suggestion
export default FilePathwithActionHeader;
```
can we get the filename also updated to action header

File: packages/terra-collapsible-menu-view/src/CollapsibleMenuView.jsx [hunk 2]
```diff
@@ -114,9 +121,21 @@ class CollapsibleMenuView extends React.Component {
...
+    this.isContainerWidthUsedtoResize = false;
+    // if no wrapper is used on top of collapsiblemenuview, use observer width value to calculate availableWidth space,
+    // or use menuButtonContainerWidth value to calculate availableWidth space
+    if (width < menuButtonContainerWidth) {
+      availableWidth = menuButtonContainerWidth - menuButtonWidth;
+      this.isContainerWidthUsedtoResize = true;
+    }
+    // to calculate available space when resized only when menuButtonContainerWidth is used
+    if ((window.innerWidth - menuButtonWidth) < availableWidth) {
+      availableWidth = window.innerWidth - menuButtonWidth;
```
Comment (supreethmr) [inline packages/terra-collapsible-menu-view/src/CollapsibleMenuView.jsx:137]: will this be having -ve value ..?? I think we can use Math.Abs() if we are not sure edge case where it could be -ve

File: packages/terra-framework-docs/src/terra-dev-site/test/file-path/FilePathwithClinicalHeader.test.jsx (hunk 1 above)
Comment (MadanKumarGovindaswamy) [reply packages/terra-framework-docs/src/terra-dev-site/test/file-path/FilePathwithClinicalHeader.test.jsx]: Updated. 
Commit: a09f4db04f5e8dd3338d2e41fe1899dd66970002
Thanks

File: packages/terra-collapsible-menu-view/src/CollapsibleMenuView.jsx (hunk 2 above)
Comment (MadanKumarGovindaswamy) [reply packages/terra-collapsible-menu-view/src/CollapsibleMenuView.jsx]: Added Math.abs to avoid -ve values.
c66f453f784054782522849ab28b84da96be7655
Thanks

**Review by saket2403 — APPROVED** [review]
